packages/flutter/lib/src/rendering/mouse_tracker.dart

```diff
@@ -202,16 +202,17 @@ class MouseTracker extends ChangeNotifier {
   // In debug mode, this method uses `_debugDuringDeviceUpdate` to prevent
   // `_deviceUpdatePhase` being recursively called.
   void _deviceUpdatePhase(VoidCallback task) {
-    assert(!_debugDuringDeviceUpdate);
```

Comment:
We should not pull any debug code out of assert.

Reply:
In release mode none of debug information can be taken... with that assert app become not manageable by not responding on any action after a task failure. Thanks, I'll return assert since it's a convention for debug code.